x/staking/keeper/validator_index.go

```diff
@@ -62,25 +62,28 @@ func (k Keeper) MigrateDelegationsByValidatorIndex(ctx sdk.Context, iterationLim
 }
 
 // ParseDelegationKey parses given key and returns delagator, validator address bytes
-func ParseDelegationKey(bz []byte) (sdk.AccAddress, sdk.ValAddress, error) {
-	delAddrLen := bz[0]
-	bz = bz[1:] // remove the length byte of delegator address.
-	if len(bz) == 0 {
-		return nil, nil, fmt.Errorf("no bytes left to parse delegator address: %X", bz)
+func ParseDelegationKey(input []byte) (sdk.AccAddress, sdk.ValAddress, error) {
```

Comment:
while we're at it should we add a comment above mentioning that `input []byte` is the delegationKey with the prefix already trimmed off?

i.e. `DelegationKey                    = []byte{0x31} // key for a delegation` is not included as where this func is used we're already iterating on a prefixStore created with this

Reply:
Sure. Added a comment and threw an error if the input has the DelegationKey as a prefix.